core/src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -89,6 +93,88 @@
     public static final ParseField PARAMS_PARSE_FIELD = new ParseField("params");
 
     /**
+     * Constructs and validates a {@link Script} being parsed as XContent.
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
I'd try and more this suppression just to the variables that need it that way it isn't super wide.